lib/src/util/dart_type_utilities.dart

```diff
@@ -282,6 +261,39 @@ class InterfaceTypeDefinition {
   }
 }
 
+extension on TypeSystem {
+  bool _interfaceTypesAreUnrelated(InterfaceType type1, InterfaceType type2) {
```

Comment:
I might renamed this to `interfaceTypesAreUnrelated` but no biggie.

Reply:
Oops, I think I normally would have too 🤣  copy-paste error.